src/React.purs

```diff
@@ -332,7 +331,7 @@ class ReactPropFields (required :: # Type) (given :: # Type)
 
 type ReservedReactPropFields r =
   ( key :: String
-  , ref :: SyntheticEventHandler (Nullable ReactRef)
+  , ref :: Ref DOMRef
```

Comment:
I don't think this is correct. A class component isn't going to have a `DOMRef` since it will be pointing to the instance. As is, the only thing that would make sense to me is `Foreign`. Since we are redoing refs anyway, maybe there's a better way to handle this side of the interface in a typed way... I'm not sure though.

Reply:
You're right. It could be the instance. I wouldn't mind `Foreign`, but what if we made an opaque type `ForwardRef` instead? I am not sure if there is a better way to hand this, but I am open to ideas.